core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -558,9 +558,9 @@ static void prepareShrinkIndexSettings(ClusterState currentState, Set<String> ma
         indexSettingsBuilder
             // we can only shrink to 1 index so far!
             .put("index.number_of_shards", 1)
-            // we use "i.r.a.include" rather than "i.r.a.require" since it's allows one of the nodes holding an
-            // instanceof all shards.
-            .put("index.routing.allocation.include._id",
+            // we use "i.r.a.initial_recovery" rather than "i.r.a.require|include" since we want the replica to allocate right away
+            // once we are allocated.
+            .put("index.routing.allocation.initial_recovery._id",
```

Comment:
Ok, I understand that. I also don't like the added complexity of having this removal code in another place...